src/internal_events/kubernetes/instrumenting_watcher.rs

```diff
@@ -0,0 +1,43 @@
+use super::InternalEvent;
+use metrics::counter;
+use std::fmt::Debug;
+
+#[derive(Debug)]
+pub struct WatchRequestInvoked;
+
+impl InternalEvent for WatchRequestInvoked {
+    fn emit_metrics(&self) {
+        counter!("k8s_watch_request_invoked", 1);
+    }
+}
+
+#[derive(Debug)]
+pub struct WatchRequestInvocationFailed<E> {
+    pub error: E,
+}
+
+impl<E: Debug> InternalEvent for WatchRequestInvocationFailed<E> {
+    fn emit_logs(&self) {
+        error!(message = "watch invocation failed", ?self.error);
```

Comment:
Is it worth rate limiting these errors?

Reply:
The requests thmeselves should be rate limted, so rate limiting logs shound’t be required. In fact, if we don’t rate linit the logs it might be easier to spot that there are too much requests attempted.

Reply:
Any log in the hot path should be rate-limited regardless. So if this is not in the hot path it should be fine. As a safe gaurd I'm not opposed to adding a 5 or 1 second rate limit.

Reply:
It's not in the hot path, watch requests are performed in the background loop at most every second (should add exponential backoff there btw, forgot about it) and should be much colder than the event processing.

A safeguard is a good idea, I'll add it.